packages/pretty-format/src/__tests__/immutable.test.js

```diff
@@ -438,56 +435,104 @@ describe('Immutable.OrderedMap plugin', () => {
+      {
+        min: true,
+      },
+    );
   });
 
   it('supports object elements {min: false}', () => {
     expect(Immutable.OrderedMap({key: {a: 1, b: 2, c: 3}})).toPrettyPrintTo(
-      'Immutable.OrderedMap {\n  key: Object {\n    "a": 1,\n    "b": 2,\n    "c": 3,\n  },\n}',
+      'Immutable.OrderedMap {\n  "key": Object {\n    "a": 1,\n    "b": 2,\n    "c": 3,\n  },\n}',
     );
   });
 
-  it('supports React components {min: true}', () => {
-    const reactComponent = React.createElement('Mouse', null, 'Hello World');
+  it('supports React elements {min: true}', () => {
+    const reactElement = React.createElement('Mouse', null, 'Hello World');
     expect(
-      Immutable.OrderedMap({a: reactComponent, b: reactComponent}),
+      Immutable.OrderedMap({a: reactElement, b: reactElement}),
     ).toPrettyPrintTo(
-      'Immutable.OrderedMap {a: <Mouse>Hello World</Mouse>, b: <Mouse>Hello World</Mouse>}',
+      'Immutable.OrderedMap {"a": <Mouse>Hello World</Mouse>, "b": <Mouse>Hello World</Mouse>}',
       {min: true},
     );
   });
 
-  it('supports React components {min: false}', () => {
-    const reactComponent = React.createElement('Mouse', null, 'Hello World');
+  it('supports React elements {min: false}', () => {
+    const reactElement = React.createElement('Mouse', null, 'Hello World');
     expect(
-      Immutable.OrderedMap({a: reactComponent, b: reactComponent}),
+      Immutable.OrderedMap({a: reactElement, b: reactElement}),
     ).toPrettyPrintTo(
-      'Immutable.OrderedMap {\n  a: <Mouse>\n    Hello World\n  </Mouse>,\n  b: <Mouse>\n    Hello World\n  </Mouse>,\n}',
+      'Immutable.OrderedMap {\n  "a": <Mouse>\n    Hello World\n  </Mouse>,\n  "b": <Mouse>\n    Hello World\n  </Mouse>,\n}',
     );
   });
+
+  it('supports non-string keys', () => {
+    const val = Immutable.OrderedMap([
+      [false, 'boolean'],
+      ['false', 'string'],
+      [0, 'number'],
+      ['0', 'string'],
+      [null, 'null'],
+      ['null', 'string'],
+      [undefined, 'undefined'],
+      ['undefined', 'string'],
+      [Symbol('description'), 'symbol'],
+      ['Symbol(description)', 'string'],
+      [['array', 'key'], 'array'],
+      [{key: 'value'}, 'object'],
+      [Immutable.Map({key: 'value'}), 'immutable map'],
+    ]);
+    const expected = [
+      'Immutable.OrderedMap {',
+      '  false: "boolean",',
+      '  "false": "string",',
+      '  0: "number",',
+      '  "0": "string",',
+      '  null: "null",',
+      '  "null": "string",',
+      '  undefined: "undefined",',
+      '  "undefined": "string",',
+      '  Symbol(description): "symbol",',
+      '  "Symbol(description)": "string",',
+      '  Array [',
+      '    "array",',
+      '    "key",',
+      '  ]: "array",',
+      '  Object {',
+      '    "key": "value",',
+      '  }: "object",',
+      '  Immutable.Map {',
+      '    "key": "value",',
+      '  }: "immutable map",',
+      '}',
```

Comment:
I think at some point we should switch such tests to template literals plus [`dedent`](https://www.npmjs.com/package/dedent)